src/server.ts

```diff
@@ -223,10 +230,47 @@ export class AceBaseServer extends SimpleEventEmitter {
         // Create websocket server
         addWebsocketServer(routeEnv);
 
+        // Register all the routes for the app
+        app.use(this.config.route, router);
+
         // Last but not least, add 404 handler
         // DISABLED because it causes server extension routes through server.extend (see above) not be be executed
         // add404Middleware(routeEnv);
 
+        // Offload shutdown control to an external server
+        if (env.server) {
+            this.pause = () => { throw new AceBaseExternalServerError() };
```

Comment:
For code readability I think it is better to add `env.server` checks to the existing methods instead of overriding them